### PR TITLE
chore: allow depending on arrow version 43.0.0.

### PR DESCRIPTION
**Pull request by youngsofun**

### CODE REVIEW

**Comment by youngsofun** [comment]
cc @Xuanwo

File: Cargo.toml [hunk 1]
```diff
@@ -16,8 +16,8 @@ io_parquet = ["arrow_schema", "dep:parquet", "dep:arrow-array"]
 anyhow = "1"
 async-trait = "0.1"
 apache-avro = { version = "0.15", features = ["derive"] }
-arrow-array = { version = ">=40, <43", optional = true }
-arrow-schema = { version = ">=40, <43", optional = true }
+arrow-array = { version = ">=40, <44", optional = true }
+arrow-schema = { version = ">=40, <44", optional = true }
```
Comment (Xuanwo) [inline Cargo.toml:20]: maybe `parquet` should also be updated.

**Review by Xuanwo — APPROVED** [review]
Thanks a lot!

**Comment by youngsofun** [comment]
```
error: package `half v2.3.1` cannot be built because it requires rustc 1.70 or newer, while the currently active rustc version is 1.69.0
Either upgrade to rustc 1.70 or newer, or use
cargo update -p half@2.3.1 --precise ver
where `ver` is the latest version of `half` supporting rustc 1.69.0
```

**Pull request merged**